src/sentry/deletions/defaults/group.py

```diff
@@ -260,7 +260,40 @@ def delete_project_group_hashes(project_id: int) -> None:
     delete_group_hashes(project_id, issue_platform_group_ids)
 
 
-def update_group_hash_metadata_in_batches(hash_ids: Sequence[int]) -> int:
+def update_group_hash_metadata_in_batches(hash_ids: Sequence[int]) -> None:
+    """
+    Update seer_matched_grouphash to None for GroupHashMetadata rows
+    that reference the given hash_ids, in batches to avoid timeouts.
+
+    This function performs the update in smaller batches to reduce lock
+    contention and prevent statement timeouts when many rows need updating.
+    """
+    option_batch_size = options.get("deletions.group-hash-metadata.batch-size", 1000)
+    batch_size = max(1, option_batch_size)
+
+    # Collect all metadata_ids in batches to avoid large IN clauses in SELECT
+    metadata_id_batches = []
+    collected_ids = set()
+    while True:
+        batch_metadata_ids = list(
+            GroupHashMetadata.objects.filter(seer_matched_grouphash_id__in=hash_ids)
```

Comment:
Note that `hash_ids` is always a maximum of 10 hashes. It's the number of rows we select which can be massive.